### ISSUE TITLE
Connect in Query editor / Notebook automatically adds to the Servers list

### ISSUE THREAD

**Issue by kevcunnane** [microsoft/azuredatastudio#5769]
Issue Type: <b>Bug</b>

- Open a new query or notebook
- Connect to a server + database combo, that is not currently in the Servers list in the Connection view

Expected:
- Editor connects, Servers view doesn't change
Actual:
- A new item is added to Connections view and automatically connected!

This is happening in master, must have recently regressed. 

@abist @anthonydresser did you make changes that could have cause this? This is a June blocker.

Azure Data Studio version: azuredatastudio 1.8.0 (9ca6200018fc206d67a47229f991901a8a453781, 2017-12-15T12:00:00.000Z)
OS version: Darwin x64 18.6.0

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-8850H CPU @ 2.60GHz (12 x 2600)|
|GPU Status|2d_canvas: enabled<br>checker_imaging: disabled_off<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: enabled<br>rasterization: enabled<br>surface_synchronization: enabled_on<br>video_decode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 2, 2|
|Memory (System)|16.00GB (0.06GB free)|
|Process Argv|. --remote-debugging-port=9222 --no-first-run --no-default-browser-check --inspect=5875 --no-cached-data|
|Screen Reader|no|
|VM|0%|
</details>Extensions: none

**Comment by anthonydresser** [comment]
Can you add more specific repro steps? If i have server "A" with Database "B" in my tree, connect an editor that isn't connected and use server "A" database "master" this is not reproing for me.

**Comment by anthonydresser** [comment]
Even if I connect to a server in general that I don't have in my tree I can't repro

**Comment by anthonydresser** [comment]
In attempting to repro I found a different but where if you don't have any connections in your tree and you connect through the editor to a server the "add server" button goes away in the tree but no connections show up. 

But still can't repro this exact issue.

**Comment by kevcunnane** [comment]
Repro from online discussion: Go from a notebook and especially if you have 1 connection in OE to master, then connecting to a specific DB in that will almost always add to the tree. It seems slightly non-deterministic, and can be harder to repro from New Query but does work from there too.

**Comment by kburtram** [comment]
The way I could repro from query editor is the following.

1. Connect to a server using the Connections viewlet and save the connection as normal
2. Delete the connection from Connections
3. Open a Query Editor using Ctrl-N
4. Select the connection from the Connection Dialog MRU (note that it says "Do not save" in the dialog)
5. Connect

Expected:  Query editor is connected but a connection isn't save to the tree.  Or the Server Group entry is correctly repopulated from the MRU entry and respected if changed.

Actual: Query editor connects and a connection is readded to the Connections tree.

**Comment by anthonydresser** [comment]
Looks like it was caused by https://github.com/microsoft/azuredatastudio/commit/b6e49f2bc09d87073d6d9bd6911f850f91909883 @Charles-Gagnon can you investigate.

**Comment by Charles-Gagnon** [comment]
@kevcunnane So I'm finding for Notebooks that if I add a new connection and specify "Server Group" as "Do Not Add" then it doesn't add it to the tree. If it's left as "<Default>" or a group name though then it adds it to the tree. To me this seems like expected behavior (although probably not the most intuitive). 

Are you seeing something different? I'm planning on doing some refactoring of the connection APIs next sprint to clean this up but for this release I want to see if there's any critical issues that need to be addressed now and so far this doesn't seem to be one of those. 

Karl and I were seeing behavior that seemed to be an actual issue though where even setting the Server Group to "<Do not save>" was still causing it to be saved in the OE tree - but this wasn't consistent so I'm having a difficult time coming up with a consistent repro. So if you are able to get a consistent repro for that behavior let me know.

![screen](https://user-images.githubusercontent.com/28519865/58721781-9d579900-838a-11e9-9a2a-c4dfe7ef040c.gif)

**Comment by kevcunnane** [comment]
I'm not sure, but the defaults here are clearly wrong - we're just using a standard API, and the behavior isn't expected to be where things are added to Object Explorer just because (in our case) we're changing a database in an editor. I believe the fix is in the dialog, even if it's to set the right default for any non-Object Explorer connection?

Let me know if you're blocked on anything, I do know this is a regression in experience vs. previous releases.

**Comment by kevcunnane** [comment]
@Charles-Gagnon - to be clear this is a regression in that previous releases didn't add new connections on changing the editor connection. We need to fix that - I disagree that this isn't important for this release. People change DBs in this manner all the time and the experience has gotten worse.

**Comment by anthonydresser** [comment]
If the issue is correctly b6e49f2 then this did exist in the last release.

**Comment by anthonydresser** [comment]
https://github.com/microsoft/azuredatastudio/commits/1.7.0 contains b6e49f2bc09d87073d6d9bd6911f850f91909883

**Comment by abist** [comment]
@anthonydresser That line was changed with my CMS changes which I merged into master today morning. @Charles-Gagnon and @kburtram are aware of that.

**Comment by anthonydresser** [comment]
@abist not sure what are replaying to? I was clarifying that b6e49f2 was in the last release, implying that this issue exists in the last release.

**Comment by abist** [comment]
@anthonydresser nevermind then, I misunderstood

**Comment by Charles-Gagnon** [comment]
@kevcunnane I'm able to repro this exact behavior in 1.7.0 as @anthonydresser found.

I agree this is very undesirable behavior and will try to find a small fix for this but I really want to avoid touching this area too much last minute like this. A LOT of stuff depends on this and making changes here can affect many different parts (as this issue itself shows).

**Comment by kevcunnane** [comment]
@Charles-Gagnon - very much agree. Finding similar challenges looking at some Azure section connectivity bugs where I might choose to punt them to July just to get bake time in Insiders. Luckily none are regressions, just existing bad behavior I'm finding through testing...
